src/main/java/com/lambdaschool/javaorders/controller/CustomerController.java

```diff
@@ -0,0 +1,84 @@
+package com.lambdaschool.javaorders.controller;
+
+import com.lambdaschool.javaorders.model.Customer;
+import com.lambdaschool.javaorders.service.CustomerService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import javax.validation.Valid;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+
+@RestController
+@RequestMapping("/customer")
+public class CustomerController
+{
+	@Autowired
+	private CustomerService customerService;
+
+	//localhost:2019/customer/order
+	@GetMapping(value = "/order",
+				produces = {"application/json"})
+	public ResponseEntity<?> listAllCustomers()
+	{
+		List<Customer> myCustomers = customerService.findAll();
+		return new ResponseEntity<>(myCustomers, HttpStatus.OK);
+	}
+
+	//localhost:2019/customer/{custname}
+	@GetMapping(value = "/{custname}",
+				produces = {"application/json"})
+	public ResponseEntity<?> getCustomerByName(
+			@PathVariable
+					String custname)
+	{
+		Customer c = customerService.findCustomerByName(custname);
+		return new ResponseEntity<>(c, HttpStatus.OK);
+	}
+
+	//localhost:2019/customer/data/new
+	@PostMapping(value = "/data/new",
+				 consumes = {"application/json"},
+				 produces = {"application/json"})
+	public ResponseEntity<?> addNewCustomer(@Valid
+											@RequestBody
+													Customer newCustomer) throws URISyntaxException
+	{
+		newCustomer = customerService.save(newCustomer);
+		HttpHeaders responseHeaders = new HttpHeaders();
+		URI newCustomerURI = ServletUriComponentsBuilder.fromCurrentRequest().
+				path("/custcode").
+				buildAndExpand(newCustomer.getCustcode()).toUri();
+		responseHeaders.setLocation(newCustomerURI);
+
+		return new ResponseEntity<>(null, responseHeaders, HttpStatus.CREATED);
+	}
+
+	//localhost:2019/customer/data/customer/update/{custcode}
+	@PutMapping(value = "/data/customer/update/{custcode}")
+	public ResponseEntity<?> updateCustomer(
+			@RequestBody
+					Customer updateCustomer,
+			@PathVariable
+					long custcode)
+	{
+		customerService.update(updateCustomer, custcode);
+		return new ResponseEntity<>(updateCustomer, HttpStatus.OK);
+	}
```

Comment:
Returning the Customer update object might look confusing, since if not all field values are supplied, you'll end up with a bunch of fields set to default values like `0`, `0.0`, `null`, etc.

Instead, you can either choose to not return an object in your `ResponseEntity` or you can set up a `findById` method in your services to get the customer with the updated information.